megamek/src/megamek/client/ui/swing/AdvancedSearchDialog.java

```diff
@@ -666,6 +721,15 @@
                 enableOperationButtons();
                 disableSelectionButtons();
             }
+            row = tblWeaponType.getSelectedRow();
+            if (row >= 0) {
+                int qty = Integer.parseInt((String)tblWeaponType.getValueAt(row, WeaponClassTableModel.COL_QTY));
```

Comment:
This will throw an exception if the relevant textbox doesn't have an actual number in it - is this possibility accounted for somehow?

Reply:
I copied the same style as the equipment and weapon tables above it. They also don't check for the exception, probably because the value comes from a drop-down box, and so should always be present. The same is true for this table.

Reply:
Fair enough.